LethalAPI.Core/Patches/HarmonyTools/InstructionsToSearch.cs

```diff
@@ -0,0 +1,106 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="InstructionsToSearch.cs" company="LethalAPI Modding Community">
+// Copyright (c) LethalAPI Modding Community. All rights reserved.
+// Licensed under the GPL-3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace LethalAPI.Core.Patches.HarmonyTools;
+
+using System.Collections.Generic;
+using System.Linq;
+
+using OpCode = System.Reflection.Emit.OpCode;
+
+/// <summary>
+/// A helper tool to find code at an index.
+/// </summary>
+// ReSharper disable MemberCanBePrivate.Global
+public readonly struct InstructionsToSearch
```

Comment:
Is this necessary considering the Harmony CodeMatcher feature? Seems to do the same, just with built-in Harmony support?

Reply:
I don't know. I needed a custom implmentation to detect for-loops, which may not be possible with harmony, due to the complex compilation of `for loops` in ILCode. I will have to investigate

Reply:
All good. Please add a TODO or issue to check this in the future.